README.md

```diff
@@ -110,106 +109,95 @@ Written in __lowerCamelCase__. For example `setValue`.
 
 Written in __lowerCamelCase__.
 
-Static fields should be written in __uppercase__, with an underscore separating
-words:
+Static fields should be written in __uppercase__, with an underscore separating words:
 
-```java
-public static final int THE_ANSWER = 42;
+```kotlin
+public static val THE_ANSWER = 42
 ```
 
-As distasteful as it is, field naming should follow the Android source code
-naming conventions:
+As distasteful as it is, field naming should follow the Android source code naming conventions:
 
 - Non-public, non-static field names start with an `m`.
 - Static field names start with an `s`.
 
 For example:
 
-```java
-public class MyClass {
-  public static final int SOME_CONSTANT = 42;
-  public int publicField;
-  private static MyClass sSingleton;
-  int mPackagePrivate;
-  private int mPrivate;
-  protected int mProtected;
+```kotlin
+open class MyClass {
+  public static val SOME_CONSTANT = 42
+  public publicField: Int
+  private static let sSingleton = MyClass()
+  mPackagePrivate: Int?
+  private mPrivate: Int?
+  protected mProtected: Int? 
 }
 ```
 
-> __Note:__ You can set Android Studio to follow this convention. See this SO
-> link for details http://stackoverflow.com/questions/22732722/intellij-android-studio-member-variable-prefix
+**TODO: UPDATE**
+> __Note:__ You can set Android Studio to follow this convention. See this SO link for details http://stackoverflow.com/questions/22732722/intellij-android-studio-member-variable-prefix
 
 ### Variables & Parameters
 
 Written in __lowerCamelCase__.
 
-Single character values to be avoided except for temporary looping variables.
+Single character values must be avoided, except for temporary looping variables.
 
 ### Misc
 
 In code, acronyms should be treated as words. For example:
 
 __BAD:__
 
-```java
+```kotlin
 XMLHTTPRequest
-String URL
+URL: String? 
 findPostByID
 ```
 __GOOD:__
 
-```java
+```kotlin
 XmlHttpRequest
-String url
+url: String
 findPostById
 ```
 
 ## Declarations
 
 ### Access Level Modifiers
 
-Access level modifiers should be explicitly defined for classes, methods and
-member variables.
+Access level modifiers should be explicitly defined for classes, methods and member variables.
 
 ### Fields & Variables
 
 Prefer single declaration per line.
 
-__BAD:__
-
-```java
-String username, twitterHandle;
-```
-
 __GOOD:__
 
-```java
-String username;
-String twitterHandle;
+```kotlin
+username: String
+twitterHandle: String
 ```
 
 ### Classes
 
-Exactly one class per source file, although inner classes are encouraged where
-scoping appropriate.
+Exactly one class per source file, although inner classes are encouraged where scoping appropriate.
 
 
```

Comment:
New section:

## Data Type Objects

Prefer `data` classes for simple data holding objects.

__BAD:__
```kotlin
class Person(val name: String) {
  override fun toString() : String {
    return "Person(name=$name)"
  }
}
```

__GOOD:__
```kotlin
data class Person(val name: String)
```